### PR TITLE
[PHR-3] | Dheeraj/Bipul | Add CM Patient Authenticator

### PR DESCRIPTION
**Pull request by dheeraj-p**

### CODE REVIEW

File: src/main/java/in/org/projecteka/hiu/HiuConfiguration.java [hunk 1]
```diff
@@ -430,16 +434,32 @@ public JWKSet jwkSet(GatewayProperties gatewayProperties) throws IOException, Pa
...
+        return JWKSet.load(new URL(identityServiceProperties.getJwkUrl()));
+    }
+
     @Bean
     public GatewayTokenVerifier centralRegistryTokenVerifier(@Qualifier("centralRegistryJWKSet") JWKSet jwkSet) {
         return new GatewayTokenVerifier(jwkSet);
     }
 
-    @Bean
+    @Bean("hiuUserAuthenticator")
```
Comment (ganesanarun) [inline src/main/java/in/org/projecteka/hiu/HiuConfiguration.java:430]: call it admins authenticator? I know doctors also use it.

Comment (ganesanarun) [reply src/main/java/in/org/projecteka/hiu/HiuConfiguration.java:430]: but this instance I don't think doctor role will have access to it.

File: src/main/java/in/org/projecteka/hiu/HiuConfiguration.java [hunk 2]
```diff
@@ -430,16 +434,32 @@ public JWKSet jwkSet(GatewayProperties gatewayProperties) throws IOException, Pa
...
     public Authenticator userAuthenticator(byte[] sharedSecret) throws JOSEException {
         return new UserAuthenticator(sharedSecret);
     }
 
+    @Bean({"jwtProcessor"})
+    public ConfigurableJWTProcessor<SecurityContext> getJWTProcessor() {
+        return new DefaultJWTProcessor<>();
+    }
+
+    @Bean("cmUserAuthenticator")
```
Comment (ganesanarun) [inline src/main/java/in/org/projecteka/hiu/HiuConfiguration.java]: userAuthenticator.

File: src/main/java/in/org/projecteka/hiu/IdentityServiceProperties.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package in.org.projecteka.hiu;
+
+import lombok.*;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConfigurationProperties(prefix = "hiu.keycloak")
```
Comment (ganesanarun) [inline src/main/java/in/org/projecteka/hiu/IdentityServiceProperties.java]: hiu.userService.

File: src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java [hunk 4]
```diff
@@ -119,9 +126,20 @@ public SecurityContextRepository contextRepository(ReactiveAuthenticationManager
...
             }
+
+            if (isCMPatientRequest(exchange.getRequest().getPath().toString(), exchange.getRequest().getMethod())) {
+                return checkCMPatientToken(token);
+            }
+
             return check(token);
         }
 
+        private Mono<SecurityContext> checkCMPatientToken(String token) {
```
Comment (ganesanarun) [inline src/main/java/in/org/projecteka/hiu/SecurityConfiguration.java]: checkUserToken.

File: src/main/java/in/org/projecteka/hiu/HiuConfiguration.java (hunk 1 above)
Comment (dheeraj-p) [reply src/main/java/in/org/projecteka/hiu/HiuConfiguration.java:430]: Both doctor and admin have access to it. So I don't think it should be adminAuthenticator.

Comment (ganesanarun) [reply src/main/java/in/org/projecteka/hiu/HiuConfiguration.java:430]: But doctor will not be part of this HIU at all dude.

Comment (dheeraj-p) [reply src/main/java/in/org/projecteka/hiu/HiuConfiguration.java:430]: But this is the same HIU code which will be used for both purposes, right?

**Review by ganesanarun — APPROVED** [review]

**Pull request merged**